tests/driver.c

```diff
@@ -16,6 +17,39 @@ struct timespec start_time, end_time;
 char driver_tag_msg[NUM_TRYTES_TAG];
 ta_send_mam_res_t* res;
 
+typedef status_t (*Proxy_apis)(const iota_client_service_t* const service, const char* const obj, char** json_result);
+typedef status_t (*Proxy_apis_without_args)(const iota_client_service_t* const service, char** json_result);
+
+static struct iri_proxys_apis_s {
+  Proxy_apis api;                            // function pointer of IRI Proxy API with args
+  Proxy_apis_without_args api_without_args;  // function pointer of IRI Proxy API without args
```

Comment:
I have no idea why `Proxy_apis_without_args` has to be specified.

Reply:
Because there are two kinds of formats to call IRI Proxy API.

one format require argument to call IRI API which is wrapped in `obj`, like 
`api_get_inclusion_states(const iota_client_service_t* const service, const char* const obj, char** json_result)`;
while the other format doesn't require argument to call IRI API, like
 `api_get_node_info(const iota_client_service_t* const service, char** json_result)`.

That is the reason why I write two formats of function pointer.

Reply:
If `args` can be detected in advance, it is still feasible to unify the two usage.

Reply:
It's able to modify all IRI Proxy APIs' format to `IRI_PROXY_API(const iota_client_service_t* const service, const char* const obj, char** json_result)`, then write extra detection code in IRI Proxy API, like
```
if (obj == NULL) return;
```
However, when some interfaces call IRI Proxy API in the future, the usage of some IRI Proxy API will looks like
```
IRI_PROXY_API(service, NULL, json_result);`    // second argument NULL is 
not intuitive
```

Therefore, I split two kinds of formats in driver tests. I wonder if you think it's still proper to unify the two usage under consideration mentioned above? Thanks!

Reply:
I would stick on single form for invocation. @HowJMay, any idea?

Reply:
I am OK with the single form of these APIs.
However, the number parameters of existing APIs is varied depend on each API, so even we have a single form for invocation proxy APIs, the other APIs of TA still don't have a single form.

Reply:
Which way to you prefer, Single form for invocation or two forms for invocation in `driver.c` ? Thanks! @HowJMay

Reply:
I defer to the decision of @HowJMay. Please add corresponding `TODO` in comments.

Reply:
Copy that!